hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AbstractAmqpIntegrationTest.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.integration;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import org.eclipse.hawkbit.AmqpTestConfiguration;
+import org.eclipse.hawkbit.amqp.DmfApiConfiguration;
+import org.eclipse.hawkbit.repository.jpa.RepositoryApplicationConfiguration;
+import org.eclipse.hawkbit.repository.test.util.AbstractIntegrationTest;
+import org.junit.Before;
+import org.junit.Rule;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageProperties;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.rabbit.junit.BrokerRunning;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+
+import com.jayway.awaitility.Awaitility;
+import com.jayway.awaitility.core.ConditionFactory;
+
+@SpringApplicationConfiguration(classes = { RepositoryApplicationConfiguration.class, AmqpTestConfiguration.class,
+        DmfApiConfiguration.class })
+// Dirty context is necessary to create a new vhost and recreate all necessary
+// beans after every test class.
+@DirtiesContext(classMode = ClassMode.AFTER_CLASS)
+public abstract class AbstractAmqpIntegrationTest extends AbstractIntegrationTest {
+
+    @Rule
+    @Autowired
+    public BrokerRunning brokerRunning;
+
+    @Autowired
+    @Qualifier("dmfClient")
+    private RabbitTemplate dmfClient;
+
+    @Before
+    public void setup() {
+        dmfClient.setExchange(getExchange());
+    }
+
+    protected abstract String getExchange();
+
+    protected RabbitTemplate getDmfClient() {
+        return dmfClient;
+    }
+
+    protected ConditionFactory createConditionFactory() {
```

Comment:
This method is used in the same way over and over again:
`createConditionFactory().until(() -> {...}`

So better go for:
```
waitUntil(Callable<Boolean> condition)
```
Than it looks like this:
```
waitUntil(() -> {
     verify(replyToListener, Mockito.atLeast(1)).handleMessage(Mockito.any());
}
```